src/bin/unit_test_attribute.c

```diff
@@ -3738,7 +3738,9 @@ int main(int argc, char *argv[])
 
 			fr_dlist_init(&lines, command_line_range_t, entry);
 
-			while (fr_sbuff_adv_until(&in, SIZE_MAX, &dir_sep, '\0')) {
+			while (fr_sbuff_extend(&in)) {
+				fr_sbuff_adv_until(&in, SIZE_MAX, &dir_sep, '\0');
```

Comment:
Extend is already called in fr_sbuff_adv_until.  Could you look for/document the root cause of the crash and fix that.

Reply:
fr_sbuff_adv_until() returns 0 when the first character is a terminator (path separator in this case), so while() loop is never entered and dir remain NULL (and ':' is not handled)

fr_sbuff_adv_until() can not be used as a while() loop condition - we use the pattern as in my change in many other similar places of the code

I think the root cause is mentioned in the commit message